os/arch/arm/src/amebasmart/amebasmart_mipi.c

```diff
@@ -0,0 +1,359 @@
+	}
+
+	if (reg_val & MIPI_BIT_ERROR) {
+		reg_dphy_err = MIPIx->MIPI_DPHY_ERR;
+		MIPIx->MIPI_DPHY_ERR = reg_dphy_err;
+		if (MIPIx->MIPI_CONTENTION_DETECTOR_AND_STOPSTATE_DT & MIPI_MASK_DETECT_ENABLE) {
+			MIPIx->MIPI_CONTENTION_DETECTOR_AND_STOPSTATE_DT &= ~MIPI_MASK_DETECT_ENABLE;
+
+			MIPIx->MIPI_DPHY_ERR = reg_dphy_err;
+			MIPI_DSI_INTS_Clr(MIPIx, MIPI_BIT_ERROR);
+		}
+
+		if (MIPIx->MIPI_DPHY_ERR == reg_dphy_err) {
+			dbg("LPTX Still Error\n");
+			MIPI_DSI_INT_Config(MIPIx, ENABLE, DISABLE, FALSE);
+		}
+		reg_val &= ~MIPI_BIT_ERROR;
+	}
+
+	if (reg_val) {
+		dbg("LPTX Error Occur: 0x%x\n", reg_val);
+	}
+
+	if (reg_val2) {
+		dbg("error occured #\n");
+	}
+}
+
+static void ameabsmart_check_freq(struct lcd_data *data)
+{
+	u32 totaly = MIPI_DSI_VSA + MIPI_DSI_VBP + MIPI_DSI_VFP + data->YPixels;
+	u32 totalx = MIPI_DSI_HSA + MIPI_DSI_HBP + MIPI_DSI_HFP + data->XPixels;
+	vo_freq = totaly * totalx * MIPI_FRAME_RATE / Mhz + 4;
+	dbg("vo_freq: %d\n", vo_freq);
+	assert_param(vo_freq < 67);
+}
+
+static void amebasmart_set_clock(void)
+{
+	u32 PLLDiv = PLL_GET_NPLL_DIVN_SDM(PLL_BASE->PLL_NPPLL_CTRL1) + 2;
+	u32 PLL_CLK = XTAL_ClkGet() * PLLDiv;
+	u32 mipi_ckd = PLL_CLK / vo_freq - 1;
+	HAL_WRITE32(SYSTEM_CTRL_BASE_LP, REG_LSYS_CKD_GRP0, (HAL_READ32(SYSTEM_CTRL_BASE_LP, REG_LSYS_CKD_GRP0) & ~LSYS_MASK_CKD_MIPI) | LSYS_CKD_MIPI(mipi_ckd));
+	HAL_WRITE32(SYSTEM_CTRL_BASE_LP, REG_LSYS_CKD_GRP0, (HAL_READ32(SYSTEM_CTRL_BASE_LP, REG_LSYS_CKD_GRP0) & ~LSYS_MASK_CKD_HPERI) | LSYS_CKD_HPERI(3));
+
+	RCC_PeriphClockCmd(APBPeriph_NULL, APBPeriph_HPERI_CLOCK, ENABLE);
+	RCC_PeriphClockCmd(APBPeriph_LCDC, APBPeriph_LCDCMIPI_CLOCK, ENABLE);
+
+}
+
+static void ameabsmart_register_interrupt(void)
+{
+	InterruptDis(mipi_irq_info.num);
+	InterruptUnRegister(mipi_irq_info.num);
+	InterruptRegister((IRQ_FUN) amebasmart_mipidsi_isr, mipi_irq_info.num, (u32) mipi_irq_info.data, mipi_irq_info.priority);
+	InterruptEn(mipi_irq_info.num, mipi_irq_info.priority);
+}
+
+static bool ameabsmart_is_transfer_completed(void)
+{
+	if (send_cmd_done == 1) {
+		return true;
+	}
+	return false;
+}
+
+void mipidsi_mode_switch(bool do_enable){
+	if(do_enable){
+		MIPI_DSI_Mode_Switch(g_dsi_host.MIPIx, ENABLE);
+	}
+	else{
+		MIPI_DSI_Mode_Switch(g_dsi_host.MIPIx, DISABLE);
+	}
+}
+
+void mipidsi_acpu_reg_clear(void){
+	MIPI_DSI_INTS_ACPU_Clr(g_dsi_host.MIPIx, MIPI_DSI_INTS_ACPU_Get(g_dsi_host.MIPIx));
+}
+
+/************************************************************************************
+ * Private Functions
+ ************************************************************************************/
+static int amebasmart_mipi_attach(FAR struct mipi_dsi_host *dsi_host, FAR struct mipi_dsi_device *dsi_device)
+{
+	return 0;
+}
+
+static int amebasmart_mipi_detach(FAR struct mipi_dsi_host *dsi_host, FAR struct mipi_dsi_device *dsi_device)
+{
+	return 0;
+}
+
+static int amebasmart_mipi_transfer(FAR struct mipi_dsi_host *dsi_host, FAR const struct mipi_dsi_msg *msg)
```

Comment:
There are 2 modes under MIPI DSI, one is video mode and one is command mode. During initialization stage, we are in reset status which is command mode, and we will send in a table of command list for configuring the LCD. But, if we need to change some configuration during runtime (ie. sending DCS command), I think we should check current mode, if we are under video mode, then we should toggle the state before sending a command, else it might not work.
So, we should add a check current DSI mode, and if it's video mode, we should invoke `MIPI_DSI_Mode_Switch(MIPI, DISABLE);` before `amebasmart_mipidsi_send_cmd`, and `MIPI_DSI_Mode_Switch(MIPI, ENABLE);` after sending command.

Reply:
ok, will add the check, 
I have a question: Will there be any case that during runtime we will change the LCD configuration. As far as I know, LCD configuration should required to set in bootup. Why anyone will change the configuration when it's running?

Reply:
It depends on application usage, but I believe more request will come in along the way, thus we should get prepared here.
From my understanding, there is already a scenario which is when integrating with PM module, we will have to set backlight dimming according to different state (ie. STANDBY state), and that requires sending DCS command during runtime. (ie. This is just one of the example use case, it should not be dependent to just PM module)
In general, this API should be applicable for both initialisation and runtime for configuring LCD, instead of just setting it once during bootup sequence.

Reply:
Hello, I tried to switch the mode before and after the transfer function. I observed that it requires some sleep after mode switch. If there is no sleep, then there is a crash. So, we would need to apply sleep twice for both mode switch.
So, call is like:
modeswitch from video -> cmd
some sleep
transfer
modeswitch from cmd -> video
return

I dont' get it why mode switching requires sleep. And if it's required to apply sleep then, would it be good idea to apply sleep in transfer?

Reply:
Hello @namanjain7 , I'm not sure why is there a crash for mode switch. Could you attach the crash log via email or some other platforms?
Also, other than mode switch, I think we should also do a 
`MIPI_DSI_INT_Config(MIPIx, DISABLE, ENABLE, FALSE);` to enable the interrupt for command mode.
And do a `MIPI_DSI_INT_Config(MIPIx, DISABLE, DISABLE, FALSE);` after sending command

Reply:
[error_logs.txt](https://github.com/Samsung/TizenRT/files/15219893/error_logs.txt)

This file attached contains error logs.

Reply:
Error in hv_isr

Reply:
Hi @namanjain7 , could you please revert this request first? I mean you can ignore the runtime sending DCS command, just leave it as the initial working codes.
I will add the necessary changes together with PM with MIPI in another PR.

Sorry for inconvenience caused.